R/broom_wrapper.R

```diff
@@ -999,6 +999,18 @@ do_survfit <- function(df, time, status, start_time = NULL, end_time = NULL, tim
   # NSE column name and it throws an evaluation error
   # without it
   if (is.null(substitute(time))) {
+    start_time_col <- col_name(substitute(start_time))
+    df[[start_time_col]] <- as.Date(df[[start_time_col]]) # convert to Date in case it is POSIXct.
+    if (!is.null(substitute(end_time))) { # if end_time exists, fill NA with today()
+      end_time_col <- col_name(substitute(end_time))
+      df[[end_time_col]] <- as.Date(df[[end_time_col]]) # convert to Date in case it is POSIXct.
+      df[[end_time_col]] <- impute_na(df[[end_time_col]] ,type = "value", val=lubridate::today())
+    }
+    else { # if end_time does not exist, create .end_time column with value of today()
+      end_time_col <- ".end_time"
```

Comment:
you can use and it's safer not to override existing column `avoid_conflict(colnames(df), ".end_time")`